src/parser/src/first_pass/prop_controller.rs

```diff
@@ -132,6 +158,18 @@ impl PropController {
                 is_player_prop: true,
             });
         }
+        if let Some(wanted_state) = self.wanted_prop_states.get(&("active_weapon_original_owner".to_string())) {
```

Comment:
I don't like how all of this code works with a bunch of if statements. Is there a reason all of the custom props are not stored as a map and iterated over keys conditionally (if the prop is  in `wanted_player_props`) pushing each prop's metadata (value in the map)  into `prop_infos`?

I can refactor if the current approach is just a draft.